graphql/util.go

```diff
@@ -1224,3 +1224,60 @@ func collapseCommit(ctx context.Context, mainlineCommits MainlineCommits, mainli
 		mainlineCommitVersion.RolledUpVersions = []*restModel.APIVersion{&apiVersion}
 	}
 }
+
+func getProjectPermissionLevel(projectPermission ProjectPermission, access AccessLevel) (requiredPermission string, requiredLevel int, err error) {
+	var permission string
+	var level int
+
+	switch projectPermission {
+	case ProjectPermissionSettings:
+		permission = evergreen.PermissionProjectSettings
+		if access == AccessLevelEdit {
+			level = evergreen.ProjectSettingsEdit.Value
+		} else if access == AccessLevelView {
+			level = evergreen.ProjectSettingsView.Value
+		} else {
```

Comment:
projectSettingsNone is a valid setting -- I think we should just return the value and not error (same with below)

Reply:
I think it might be better to error to let the implementer know that they specified an incorrect permission; for example, if they specified `VIEW` access for `permission_patches` (no such permission exists). When applying directives, we will never specify a `NONE` access, so I think we wouldn't expect it to be valid

Reply:
Ah I suppose I get that -- we're passing in the access level we're looking for, not just what the user has. Thanks for explaining!